minecode/model_utils.py

```diff
@@ -30,6 +30,11 @@
 DEFAULT_PIPELINES = (
     'scan_single_package',
     'fingerprint_codebase',
+)
+
+# These are the list of supported addon pipelines to run when we scan a Package for
+# indexing.
+SUPPORTED_ADDON_PIPELINES = (
```

Comment:
Can you give me an example of when these pipelines would be run when indexing a package. I see how the default pipelines would be run, but where would we add the symbol collecting pipelines when we index a package?

Reply:
@JonoYang 
Suppose we receive this request: `/api/collect/?purl=pkg:npm/foo@1.2.3&addon_pipelines=collect_symbols`. 

First, the **CollectPackageSerializer** will validate whether `collect_symbols` is a valid pipeline or not using `SUPPORTED_ADDON_PIPELINES`. Then, will add this package to the scan queue using add_package_to_scan_queue, with the `pipeline` argument being `DEFAULT_PIPELINES` + ( 'collect_symbols' ).

Reply:
I see, do we currently use this anywhere? Otherwise, I think the code looks good unless there's something you want to change.

Reply:
@JonoYang No more change from my side, it's ready for merge.